From 56490c0e8561d5fe7130ee9dc547c91b38ad5c08 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Sat, 24 Feb 2018 16:48:18 +0800
Subject: [PATCH] ARM: dts: rockchip: use regulator for otg vbus on
 rk3128-fireprime

This patch uses a fixed-regulator instead of GPIO pin for
otg vbus power on rk3128-fireprime board.

Change-Id: I5b1aa112ec8e1f07d9a6c2f7d0b349ab30331913
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3128-fireprime.dts | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3128-fireprime.dts b/arch/arm/boot/dts/rk3128-fireprime.dts
index 0aa459285f86..1f1283957646 100644
--- a/arch/arm/boot/dts/rk3128-fireprime.dts
+++ b/arch/arm/boot/dts/rk3128-fireprime.dts
@@ -103,6 +103,17 @@
 		regulator-name = "vcc_host_5v";
 		regulator-always-on;
 	};
+
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio0 RK_PD6 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
 };
 
 &cpu0 {
@@ -394,6 +405,10 @@
 		host_vbus_drv: host-vbus-drv {
 			rockchip,pins = <3 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <0 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
 	};
 };
 
@@ -420,9 +435,9 @@
 
 &u2phy {
 	status = "okay";
-	otg-vbus-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
 
 	u2phy_otg: otg-port {
+		vbus-supply = <&vcc_otg_vbus>;
 		status = "okay";
 	};
 
-- 
2.35.3

